src/main/webapp/app/shared/exercise-filter/exercise-filter-modal.component.ts

```diff
@@ -0,0 +1,229 @@
+        const filterOption: ExerciseCategoryFilterOption = event.item;
+        filterOption.searched = true;
+        this.updateCategoryOptionsStates();
+        this.model = undefined; // Clear the input field after selection
+    }
+
+    removeItem(item: ExerciseCategoryFilterOption): () => void {
+        return () => {
+            item.searched = false;
+            this.updateCategoryOptionsStates();
+        };
+    }
+
+    applyFilter(): void {
+        if (!this.sidebarData?.groupedData) {
+            return;
+        }
+
+        const appliedFilterDetails = this.getAppliedFilterDetails();
+        for (const groupedDataKey in this.sidebarData.groupedData) {
+            this.sidebarData.groupedData[groupedDataKey].entityData = this.sidebarData.groupedData[groupedDataKey].entityData.filter((sidebarElement) =>
+                satisfiesFilters(sidebarElement, appliedFilterDetails),
+            );
+        }
+        this.sidebarData.ungroupedData = this.sidebarData.ungroupedData?.filter((sidebarElement: SidebarCardElement) => satisfiesFilters(sidebarElement, appliedFilterDetails));
+
+        this.filterApplied.emit({
+            filteredSidebarData: this.sidebarData,
+            appliedExerciseFilters: this.exerciseFilters,
+            isFilterActive: this.isFilterActive(appliedFilterDetails),
+        });
+
+        this.closeModal();
+    }
+
+    private getAppliedFilterDetails(): FilterDetails {
+        return {
+            searchedTypes: this.getSearchedTypes(),
+            selectedCategories: this.getSelectedCategories(),
+            searchedDifficulties: this.getSearchedDifficulties(),
+            isScoreFilterApplied: isRangeFilterApplied(this.achievedScore),
+            isPointsFilterApplied: isRangeFilterApplied(this.achievablePoints),
+            achievedScore: this.achievedScore,
+            achievablePoints: this.achievablePoints,
+        };
+    }
+
+    private getSearchedTypes(): ExerciseType[] | undefined {
+        return this.typeFilter?.options.filter((type) => type.checked).map((type) => type.value);
+    }
+
+    private getSelectedCategories(): ExerciseCategory[] {
+        return this.selectedCategoryOptions.map((categoryOption: ExerciseCategoryFilterOption) => categoryOption.category);
+    }
+
+    private getSearchedDifficulties(): DifficultyLevel[] | undefined {
+        return this.difficultyFilter?.options.filter((difficulty) => difficulty.checked).map((difficulty) => difficulty.value);
+    }
+
+    private isFilterActive(filterDetails: FilterDetails): boolean {
+        return (
+            !!filterDetails.selectedCategories.length ||
+            !!filterDetails.searchedTypes?.length ||
+            !!filterDetails.searchedDifficulties?.length ||
+            filterDetails.isScoreFilterApplied ||
+            filterDetails.isPointsFilterApplied
+        );
+    }
```

Comment:
Simplified `getAppliedFilterDetails` by adding methods for the parameters instead of inline calculations